src/delegation/Delegation.ts

```diff
@@ -77,11 +76,9 @@ export default abstract class DelegationBaseNode
     const queryResults: CodecWithId[] = await fetchChildren(childIds)
     const children: DelegationNode[] = queryResults
       .map((codec: CodecWithId) => {
-        const decoded = this.decodeChildNode(codec.codec)
-        if (decoded) {
-          decoded.id = codec.id
-        }
-        return decoded
+        const decoded = decodeDelegationNode(codec.codec)
+        if (decoded) return { ...decoded, id: codec.id } as DelegationNode
```

Comment:
this should probably use the constructor - otherwise we'd circumvent the constructor input checks

Reply:
I agree!

Reply:
fixed that now!